File: php/structure-in-rte/ImplementLinkResolver.php
<?php
// DocSection: structure_in_rte_implement_link_resolver
// Tip: Find more about PHP SDKs at https://docs.kontent.ai/php
class CustomContentLinkUrlResolver implements ContentLinkUrlResolverInterface
{
    public function resolveLinkUrl($link)
    {
        // Resolves URLs to content items based on the Article content type
        if ($link->contentTypeCodeName == "my_article") {
            return "/articles/". $link->urlSlug;
        }

        // TODO: Add the rest of the resolver logic
    }

    public function resolveBrokenLinkUrl()
    {
        // Resolves URLs to unavailable content items
        return "/404";
    }    
}
// EndDocSection
?>

File: java/structure-in-rte/ImplementLinkResolver.java
// DocSection: structure_in_rte_implement_link_resolver
// Tip: Find more about Java SDK at https://kontent.ai/learn/java
public class CustomContentLinkUrlResolver implements ContentLinkUrlResolver {
    
    @Override
    String resolveLinkUrl(Link link) {
        // Resolves URLs to content items based on the Article content type
        if ("my_article".equals(link.getCodename())) {
            return String.format("/articles/%s", link.getUrlSlug());
        }
    }
}

public class CustomBrokenContentLinkUrlResolver implements BrokenLinkUrlResolver {

    @Override
    String resolveBrokenLinkUrl() {
        // Resolves URLs to unavailable content items
        return "/404";
    }
}
// EndDocSection